packages/babel-plugin/src/class-names/index.tsx

```diff
@@ -7,6 +7,43 @@ import { buildCss, getItemCss } from '../utils/css-builders';
 import { Metadata } from '../types';
 import { CSSOutput } from '../utils/types';
 
+/**
+ * Handles style prop value. If variables are present it will replace its value with it
+ * otherwise will add undefined.
+ *
+ * @param variables CSS variables prop to be placed as inline styles
+ * @param path Any Expression path
+ */
+const handleStyleProp = (variables: CSSOutput['variables'], path: NodePath<t.Expression>) => {
+  const styleValue = variables.length
+    ? t.objectExpression(buildCssVariablesProp(variables))
+    : t.identifier('undefined');
+
+  path.replaceWith(styleValue);
+};
+
+/**
+ * Will recursively checks if type (`css` or `style`) coming from destructuring.
+ *
+ * @param type `css` or `style` type
+ * @param node Any Expression node
+ */
+const isTypeComingFromDestructuring = (
```

Comment:
we probably want to put this into `ast.tsx`

Reply:
oooo cool

Reply:
thanks!